instrumentation/github.com/aws/aws-sdk-go-v2/otelaws/aws_test.go

```diff
@@ -0,0 +1,162 @@
+		responseBody       []byte
+		expectedError      string
+		expectedRequestID  string
+		expectedStatusCode int
+	}{
+		"invalidChangeBatchError": {
+			responseStatus: 500,
+			responseBody: []byte(`<?xml version="1.0" encoding="UTF-8"?>
+		<InvalidChangeBatch xmlns="https://route53.amazonaws.com/doc/2013-04-01/">
+		  <Messages>
+		    <Message>Tried to create resource record set duplicate.example.com. type A, but it already exists</Message>
+		  </Messages>
+		  <RequestId>b25f48e8-84fd-11e6-80d9-574e0c4664cb</RequestId>
+		</InvalidChangeBatch>`),
+			expectedError:      "Error",
+			expectedRequestID:  "b25f48e8-84fd-11e6-80d9-574e0c4664cb",
+			expectedStatusCode: 500,
+		},
+
+		"standardRestXMLError": {
+			responseStatus: 404,
+			responseBody: []byte(`<?xml version="1.0"?>
+		<ErrorResponse xmlns="http://route53.amazonaws.com/doc/2016-09-07/">
+		  <Error>
+		    <Type>Sender</Type>
+		    <Code>MalformedXML</Code>
+		    <Message>1 validation error detected: Value null at 'route53#ChangeSet' failed to satisfy constraint: Member must not be null</Message>
+		  </Error>
+		  <RequestId>1234567890A</RequestId>
+		</ErrorResponse>
+		`),
+			expectedError:      "Error",
+			expectedRequestID:  "1234567890A",
+			expectedStatusCode: 404,
+		},
+
+		"Success response": {
+			responseStatus: 200,
+			responseBody: []byte(`<?xml version="1.0" encoding="UTF-8"?>
+		<ChangeResourceRecordSetsResponse>
+   			<ChangeInfo>
+      		<Comment>mockComment</Comment>
+      		<Id>mockID</Id>
+   		</ChangeInfo>
+		</ChangeResourceRecordSetsResponse>`),
+			expectedStatusCode: 200,
+		},
+	}
+
+	for name, c := range cases {
+		server := httptest.NewServer(http.HandlerFunc(
+			func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(c.responseStatus)
+				_, err := w.Write(c.responseBody)
+				if err != nil {
+					t.Fatal(err)
+				}
+			}))
+		defer server.Close()
+
+		t.Run(name, func(t *testing.T) {
+			sr := new(oteltest.SpanRecorder)
+			provider := oteltest.NewTracerProvider(oteltest.WithSpanRecorder(sr))
+
+			svc := route53.NewFromConfig(aws.Config{
+				Region: "us-east-1",
+				EndpointResolver: aws.EndpointResolverFunc(func(service, region string) (aws.Endpoint, error) {
+					return aws.Endpoint{
+						URL:         server.URL,
+						SigningName: "route53",
+					}, nil
+				}),
+				Retryer: func() aws.Retryer {
+					return aws.NopRetryer{}
+				},
+			})
+			_, err := svc.ChangeResourceRecordSets(context.Background(), &route53.ChangeResourceRecordSetsInput{
+				ChangeBatch: &types.ChangeBatch{
+					Changes: []types.Change{},
+					Comment: aws.String("mock"),
+				},
+				HostedZoneId: aws.String("zone"),
+			}, func(options *route53.Options) {
+				AppendMiddlewares(
+					&options.APIOptions, WithTracerProvider(provider), WithPropagators(b3.B3{}))
+			})
+
+			spans := sr.Completed()
+			assert.Len(t, spans, 1)
+			span := spans[0]
+
+			if span.Name() != "Route 53" {
+				t.Fatalf("Span Name is not 'Route 53'.")
```

Comment:
Can we add both the expected and actual values to the outputted message for these first three tests to help with debugging?

Reply:
yes